packages/http.go

```diff
@@ -15,11 +15,11 @@ import (
 	"github.com/elastic/package-registry/util"
 )
 
-func ServePackage(w http.ResponseWriter, r *http.Request, p *Package) {
+// ServeLocalPackage is used by artifactsHandler to serve packages and signatures.
+func ServeLocalPackage(w http.ResponseWriter, r *http.Request, p *Package, packagePath string) {
 	span, _ := apm.StartSpan(r.Context(), "ServePackage", "app")
 	defer span.End()
 
-	packagePath := p.BasePath
```

Comment:
Couldn't we continue using `p.BasePath` as the `packagePath`? Having to pass the package and its path can be error prone.

Reply:
Keep in mind that the same function is used to serve signatures, so you can't depend on the base path.

In signatures.go:

```go
packages.ServeLocalPackage(w, r, packageList[0], packageList[0].BasePath+".sig")
```